server/coordinator/procedure/drop_partition_table.go

```diff
@@ -0,0 +1,335 @@
+			}
+			if err := p.fsm.Event(eventDropPartitionTableFinish, dropPartitionTableRequest); err != nil {
+				p.updateStateWithLock(StateFailed)
+				return errors.WithMessagef(err, "drop partition table procedure drop partition table")
+			}
+			p.updateStateWithLock(StateFinished)
+		case stateDropPartitionTableFinish:
+			if err := p.persist(ctx); err != nil {
+				return errors.WithMessage(err, "drop partition table procedure persist")
+			}
+			return nil
+		}
+	}
+}
+
+func (p *DropPartitionTableProcedure) Cancel(_ context.Context) error {
+	p.updateStateWithLock(StateCancelled)
+	return nil
+}
+
+func (p *DropPartitionTableProcedure) State() State {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	return p.state
+}
+
+func (p *DropPartitionTableProcedure) updateStateWithLock(state State) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	p.state = state
+}
+
+func (p *DropPartitionTableProcedure) persist(ctx context.Context) error {
+	meta, err := p.convertToMeta()
+	if err != nil {
+		return errors.WithMessage(err, "convert to meta")
+	}
+	err = p.storage.CreateOrUpdate(ctx, meta)
+	if err != nil {
+		return errors.WithMessage(err, "createOrUpdate procedure storage")
+	}
+	return nil
+}
+
+func (p *DropPartitionTableProcedure) convertToMeta() (Meta, error) {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	rawData := dropPartitionTableRawData{
+		ID:       p.id,
+		FsmState: p.fsm.Current(),
+		State:    p.state,
+	}
+	rawDataBytes, err := json.Marshal(rawData)
+	if err != nil {
+		return Meta{}, ErrEncodeRawData.WithCausef("marshal raw data, procedureID:%v, err:%v", p.id, err)
+	}
+
+	meta := Meta{
+		ID:    p.id,
+		Typ:   DropPartitionTable,
+		State: p.state,
+
+		RawData: rawDataBytes,
+	}
+
+	return meta, nil
+}
+
+type dropPartitionTableRawData struct {
+	ID       uint64
+	FsmState string
+	State    State
+
+	DropTableResult cluster.DropTableResult
+}
+
+type dropPartitionTableCallbackRequest struct {
+	ctx      context.Context
+	cluster  *cluster.Cluster
+	dispatch eventdispatch.Dispatch
+
+	sourceReq *metaservicepb.DropTableRequest
+
+	onSucceeded func(info cluster.TableInfo) error
+	onFailed    func(error) error
+
+	ret cluster.TableInfo
+}
+
+// 1. Drop data tables in target nodes.
+func dropDataTablesCallback(event *fsm.Event) {
+	req, err := getRequestFromEvent[*dropPartitionTableCallbackRequest](event)
+	if err != nil {
+		cancelEventWithLog(event, err, "get request from event")
+		return
+	}
+
+	for _, tableName := range req.sourceReq.PartitionTableInfo.SubTableNames {
+		err := dropTable(event, tableName)
+		if err != nil {
+			cancelEventWithLog(event, err, fmt.Sprintf("drop table, table:%s", tableName))
+		}
+	}
+}
+
+// 2. Drop partition table in target node.
+func dropPartitionTableCallback(event *fsm.Event) {
+	req, err := getRequestFromEvent[*dropPartitionTableCallbackRequest](event)
+	if err != nil {
+		cancelEventWithLog(event, err, "get request from event")
+		return
+	}
+
+	err = dropTable(event, req.sourceReq.Name)
```

Comment:
Partition table maybe opened on multiple shards, but it has only been deleted once. The opened partition table needs to be closed by reopening the shard. Please refer to the logic of creating a table.